models/patterns/pattern.go

```diff
@@ -51,6 +51,9 @@ func ProcessAnnotations(pattern *pattern.PatternFile) {
 func ProcessComponentStatus(pattern *pattern.PatternFile) {
 	components := []*component.ComponentDefinition{}
 	for _, component := range pattern.Components {
+		if component == nil {
+			continue
+		}
 		if component.Status != nil && string(*component.Status) == "ignored" {
```

Comment:
```suggestion
		hasComponentStatus := func(component *pattern.Components) {
		  return component != nil && component.Status != nil
		}
		
		if  hasComponentStatus(&component) && string(*component.Status) == "ignored" {
```
think this will be more readable

Reply:
Got it